### PR TITLE
upgrade node/python

### PR DESCRIPTION
**Pull request by chris48s**
And so the "get all our stuff onto current versions" project continues.
There is also a bit of general cleaning up odd stuff and getting things working going on in this PR.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -42,7 +42,7 @@
     "sass-export": "^2.1.2",
     "slugify": "^1.6.5"
   },
-  "dependencies": {
-    "npm": "^10.5.2"
```
Comment (chris48s) [inline package.json:46]: NPM shouldn't be a runtime dependency of this project. We shell out to `npm`, but that uses the same global install you used to install the package, not the second copy this installs into node_modules. It is quite hard to deduce the intent of this from the commit it was added in.

File: .github/workflows/run-tests.yml [hunk 2]
```diff
@@ -10,17 +10,28 @@ jobs:
...
       - name: Set up Node
         uses: actions/setup-node@v3
         with:
           node-version: 18
-      - run: npm install
-      - run: npm install -g lighthouse
+
+      - run: npm ci
+
+      - run: npm install -g lighthouse@11.7.1
```
Comment (chris48s) [inline .github/workflows/run-tests.yml:26]: To get the tests passing, I have pinned us to an older version of lighthouse ( https://github.com/DemocracyClub/design-system/commit/481d95523c9ba94742945fdd8dd7d8487b159744 was pushed just after the release of lighthouse 12). Lets address the lighthouse 11 --> lighthouse 12 migration as another issue.

File: .github/workflows/deploy.yml [hunk 3]
```diff
@@ -7,7 +7,6 @@ on:
 permissions:
   contents: read
   pages: write
-  id-token: write
```
Comment (chris48s) [inline .github/workflows/deploy.yml:10]: You don't need to generate an OIDC token to deploy to GitHub pages

File: .github/workflows/release.yaml [hunk 4]
```diff
@@ -1,53 +0,0 @@
-name: Publish Python Package
```
Comment (chris48s) [inline .github/workflows/release.yaml:1]: I've completely deleted this file on the basis I don't think this has ever run to completion. We've never pushed a version of this package to PyPI https://pypi.org/project/dc_design_system/ Also even if it had worked at one point, the auth would now be broken.
Lets just bin this for now and come back to it when we look at fixing up package publishing and trusted publisher workflow across packages.

File: .github/workflows/release.yaml [hunk 5]
```diff
@@ -1,53 +0,0 @@
...
-      - completed
-    branches:
-      - master
-    
-jobs:
-  test:
-    runs-on: ubuntu-latest
-    strategy:
-      matrix:
-        python-version: ["3.8", "3.9", "3.10", "3.11"]
```
Comment (chris48s) [inline .github/workflows/release.yaml:18]: In the remaining workflow where we are running the tests I have not bothered implementing a test matrix. Although we are using pytest in theory, none of our tests are actually running python package code. They're just shelling out to `lighthouse`. If we ever add some tests that are exercising python code, we can add a matrix there.

File: .github/workflows/run-tests.yml (hunk 2 above)
Comment (chris48s) [reply .github/workflows/run-tests.yml:26]: https://github.com/DemocracyClub/design-system/pull/142

**Comment by chris48s** [comment]
There is quite a lot of HTML in this repo, but none of it is in HTML files. It is all in markdown files which aren't syntactically valid HTML at the file level so djhtml is not our friend here. We might look at some kind of other tooling for that, but I reckon that's another job for another day. Time to stop pulling the thread.

**Review by symroe — APPROVED** [review]

**Comment by symroe** [comment]
> There is quite a lot of HTML in this repo, but none of it is in HTML files. It is all in markdown files which aren't syntactically valid HTML at the file level so djhtml is not our friend here. We might look at some kind of other tooling for that, but I reckon that's another job for another day. Time to stop pulling the thread.

Yeah, I don't think we need to worry about djhtml for non-Django projects. That project mainly deals with Django template logic, and we don't have that here anyway. If we wanted to lint the HTML here it would be better to use a more general HTML linter like `tidy`

File: .github/workflows/deploy.yml (hunk 3 above)
Comment (chris48s) [reply .github/workflows/deploy.yml:10]: So I merged this and.. fun fact
turns out we do
The deploy failed with `Error: Ensure GITHUB_TOKEN has permission "id-token: write".`

I have a bunch of repos where I write to the gh-pages branch of a repo without this permission, but having a quick scan over the docs for https://github.com/actions/deploy-pages doing it with actions/deploy-pages does it with an OIDC token :mortar_board:

Anyway.. I reverted this commit and pushed to master to fix.

**Pull request merged**